### PR TITLE
Create Communities Section on Home Page

### PR DESCRIPTION
**Pull request by jacksonmoody**
## Summary

Created a `communities.html` partial to be used on both the home page of digital.gov and on the communities page. Also refactored the communities cards to use a generic `usa-card.html` partial. Updated the styling in `_communities.scss` to accommodate  communities being displayed on the home page.

### Preview

[Link to Preview →](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.app.cloud.gov/preview/gsa/digitalgov.gov/jtm-create-homepage-communities/)

#### Before
Screenshot 2023-07-11 at 5 09 35 PM

#### After
Screenshot 2023-08-01 at 9 31 52 AM

### How To Test

1. Visit the [home page of digital.gov](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.app.cloud.gov/preview/gsa/digitalgov.gov/jtm-create-homepage-communities/)
2. Visit the [communities page of digital.gov](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/jtm-create-homepage-communities/communities/)
3. For both pages, verify that the featured communities display properly on all screen sizes

---

### Dev Checklist

- [x] PR has correct labels
- [x] A11y testing (voice over testing, meets WCAG, run axe tools)
- [x] Code is formatted properly

### CODE REVIEW

**Review by mejiaj — CHANGES_REQUESTED** [review]
@jacksonmoody this is great, do you see any issues with licensing/rights with the images?  

Visually it looks slightly disconnected from the [Communities landing](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/jtm-create-homepage-communities/communities/). Could we create a component out of this section:
https://github.com/GSA/digitalgov.gov/blob/3b1cfde66bb156c918abcd2156f69f9f8fb48a8b/themes/digital.gov/layouts/communities/list.html#L21-L66

That way we could re-use on both pages. The component would:

1. Allow modifying the grid column with a fallback value.
2. Include optional image to be set.

A quick mockup:
image

**Comment by nick-mon1** [comment]
@jacksonmoody I like the effort for this first draft. I've linked to a [figma design doc](https://www.figma.com/file/NcJkuWkiNJ6rWcImUchrqD/Communities-Component?type=design&node-id=0%3A1&mode=design&t=rPlOsxjtKjnZR5nx-1) with some visual notes about how this component can be used on the home page. 

I like the icons but I'm not sure how we can integrate them with the card box design. 

Take a look at the [collection component PR](https://github.com/GSA/digitalgov.gov/pull/6129/files#diff-368fba15b4a408d256cc802f0dd00fd0e9124075582f844dca118d06a2c59dad) for some ideas on how to create a "hugo component". We can create our community component to take in settings that @mejiaj mentions in his comments. We can set the grid display depending on the page it is used on with this approach.

**Review by nick-mon1 — CHANGES_REQUESTED** [review]
@jacksonmoody I like the effort for this first draft. I've linked to a [figma design doc](https://www.figma.com/file/NcJkuWkiNJ6rWcImUchrqD/Communities-Component?type=design&node-id=0%3A1&mode=design&t=rPlOsxjtKjnZR5nx-1) with some visual notes about how this component can be used on the home page. 

I like the icons but I'm not sure how we can integrate them with the card box design. 

Take a look at the [collection component PR](https://github.com/GSA/digitalgov.gov/pull/6129/files#diff-368fba15b4a408d256cc802f0dd00fd0e9124075582f844dca118d06a2c59dad) for some ideas on how to create a "hugo component". We can create our community component to take in settings that @mejiaj mentions in his comments. We can set the grid display depending on the page it is used on with this approach.

File: themes/digital.gov/layouts/partials/core/community-card.html
Comment (mejiaj) [inline themes/digital.gov/layouts/partials/core/community-card.html:1]: Can we create a generic usa-card that can be used for this component? Similar to how [`usa-collection`](https://github.com/GSA/digitalgov.gov/blob/main/themes/digital.gov/layouts/partials/core/usa-collection.html) is used in [`collection.html`](https://github.com/GSA/digitalgov.gov/blob/main/themes/digital.gov/layouts/partials/core/collection/collection.html)

File: themes/digital.gov/src/scss/new/_communities.scss
Comment (mejiaj) [inline themes/digital.gov/src/scss/new/_communities.scss:1]: Let's avoid utility mixins for single property values. We should favor native properties and USWDS tokens wherever possible. This way the SCSS is more readable and properties are easier to find.

For example:

```scss
  &-button {
    background-color: color("gray-10");
    border: none;
    padding: units(2);  
  }
}
```

**Review by mejiaj — CHANGES_REQUESTED** [review]
Glad we're adding more reusable components. A few comments:

1. For now, can we just show all of the communities at once, like on the landing page? We can try flexbox alignment to center the last one as an enhancement.
2. Could you create a generic `usa-card.html` next to the other USWDS components in `themes/digital.gov/layouts/partials/core`? ⚠️ If this is too high LOE we can bump this as long as there's a card in backlog. 
3. Are the logos still needed for communities? Not seeing them rendered.

File: themes/digital.gov/src/scss/new/_communities.scss
Comment (mejiaj) [inline themes/digital.gov/src/scss/new/_communities.scss:21]: Empty line before selector for readability.

```suggestion
  padding-top: units(2);
  
  &-title {
    margin-bottom: units(-5);
    margin-top: units(5);
  }
}
```

File: themes/digital.gov/src/scss/new/_communities.scss
Comment (mejiaj) [inline themes/digital.gov/src/scss/new/_communities.scss:19]: A little more spacing could help visually separate this out from hero above. Added the same units (8) as `usa-section`.

```suggestion
    margin-top: units(8);
```

**Before**
image

**After**
image

File: themes/digital.gov/src/scss/new/_communities.scss
Comment (mejiaj) [inline themes/digital.gov/src/scss/new/_communities.scss:23]: We could improve clarity by moving home specific styles to their respective page partials, like`/scss/new/home/_home.scss`.

File: themes/digital.gov/layouts/partials/core/communities.html
Comment (mejiaj) [inline themes/digital.gov/layouts/partials/core/communities.html:1]: Is it possible to re-use the same grid structure? Any modifications we can add a different selector at the top (based on location or a new variant). 

For example, parent has base class like `featured-communities` and then modifiers like `featured-communities--centered` or `featured-communities--home` that we can control differences at the top-level.

**Review by mejiaj — CHANGES_REQUESTED** [review]
Just a few minor tweaks, otherwise LGTM.

File: themes/digital.gov/src/scss/new/home/_home.scss
Comment (nick-mon1) [inline themes/digital.gov/src/scss/new/home/_home.scss:1]: Looks like this has been `_communities.scss` which makes more sense.

**Review by nick-mon1 — CHANGES_REQUESTED** [review]

File: themes/digital.gov/src/scss/new/home/_home.scss
Comment (jacksonmoody) [reply themes/digital.gov/src/scss/new/home/_home.scss:1]: > We could improve clarity by moving home specific styles to their respective page partials, like`/scss/new/home/_home.scss`.

Was trying to address this comment from @mejiaj. Agreed that it seems weird to have some styles in `home` and some in  `communities` for basically the same component, though. Maybe we can move all the styles out of `home` (except the ones for `.home`) into their respective partials just to be consistent?

File: themes/digital.gov/layouts/partials/core/usa-card.html
Comment (jacksonmoody) [reply themes/digital.gov/layouts/partials/core/usa-card.html:21]: Thanks! Addressed in [5d10e0e](https://github.com/GSA/digitalgov.gov/pull/6310/commits/5d10e0ef65e03826751a20f304634026a1c8799d)

File: themes/digital.gov/layouts/partials/core/communities.html
Comment (jacksonmoody) [reply themes/digital.gov/layouts/partials/core/communities.html:1]: Good idea! Addressed in [5d10e0e](https://github.com/GSA/digitalgov.gov/pull/6310/commits/5d10e0ef65e03826751a20f304634026a1c8799d)

File: themes/digital.gov/layouts/partials/core/communities.html
Comment (jacksonmoody) [reply themes/digital.gov/layouts/partials/core/communities.html]: Thanks! I moved the path logic outside of the partial and into this section, as it might vary depending on the context in which the partial is moved.

File: themes/digital.gov/src/scss/new/_communities.scss
Comment (mejiaj) [inline themes/digital.gov/src/scss/new/_communities.scss:56]: We should update the scope of this to make it clear it's part of the featured communities.

Consider updating the class name to something like `.dg-featured-communities .usa-card` or giving it its own class like `.dg-featured-communities__item` or `.dg-featured-communities__card`.

File: themes/digital.gov/src/scss/new/_communities.scss
Comment (mejiaj) [inline themes/digital.gov/src/scss/new/_communities.scss:16]: Following BEM naming convention this would be more like `.dg-communities__title`.

**Review by mejiaj — CHANGES_REQUESTED** [review]
Just a few comments to improve clarity and a11y. Should be good to go after.

**Review by nick-mon1 — DISMISSED** [review]

File: themes/digital.gov/layouts/partials/core/communities.html
Comment (jacksonmoody) [reply themes/digital.gov/layouts/partials/core/communities.html:23]: I like the `<header>` semantically because it matches with the `<header>`, `<section>`, `<header>`, `<section>` pattern we are using on the `/communities` page. I removed the parent `div` to avoid any conflicts.

**Review by mejiaj — DISMISSED** [review]

**Review by nick-mon1 — APPROVED** [review]

**Review by ToniBonittoGSA — COMMENTED** [review]
Closing; https://github.com/GSA/digitalgov.gov/pull/6317 already added a Communities button to the homepage.

**Comment by nick-mon1** [comment]
Work has been moved to #6872

**Pull request closed without merging**
